### ISSUE TITLE
Downstream test issues

### ISSUE THREAD

**Issue by bryevdv** [bokeh/bokeh#12223]
All of the downstream tests are currently failing, for various reasons:

## Dask

```python
______________________________ test_plot_multiple ______________________________
    @pytest.mark.skipif("not bokeh")
    @pytest.mark.skipif("not psutil")
    def test_plot_multiple():
        from dask.diagnostics.profile_visualize import visualize
        with ResourceProfiler(dt=0.01) as rprof:
            with prof:
                get(dsk2, "c")
        p = visualize(
            [prof, rprof], label_size=50, title="Not the default", show=False, save=False
        )
>       figures = [r[0] for r in p.children[1].children]
E       AttributeError: 'tuple' object has no attribute 'children'
dask/diagnostics/tests/test_profiler.py:333: AttributeError
```

and

```
E   ImportError: cannot import name 'value' from 'bokeh.models' (/usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/bokeh/models/__init__.py)
```

We can definitely put things back to fix the second. @mattpap what about the first? Dask is perhaps the most visible, important downstream project. I'd really like to take a little extra care to avoid breaking them if there is literally any possible way at all. 

## Holoviz

cc @philippjfr all test fail due to a failed import: inside Panel
```
ImportError: cannot import name 'Markup' from 'jinja2' (/usr/share/miniconda3/envs/bk-test/lib/python3.9/site-packages/jinja2/__init__.py)
```

## PandasBokeh, etc.

All tests erroring due to
```
pytest: error: unrecognized arguments: --no-success-flaky-report
```
We do `addopts = --no-success-flaky-report` in our `setup.cfg`, I guess that is causing issues?

**Comment by mattpap** [comment]
> `AttributeError: 'tuple' object has no attribute 'children'`

The source of the issue is `gridplot()` returning a `GridPlot` and not a composite layout. I don't see any workaround for this, except what they are already doing regarding `plot_width` vs `width`, etc.

> `ImportError: cannot import name 'value' from 'bokeh.models'`

Definitely not the right place for this, but it can be restored.

> We do `addopts = --no-success-flaky-report` in our `setup.cfg`, I guess that is causing issues?

Lack of flaky installed is the direct culprit. However, the real issue is with pytest and its inability to deal with conditional dependencies and configuration options.

**Comment by bryevdv** [comment]
After closer inspection think the `.children[1].children` issue is test-only and does not affect actual library code. I submitted a PR to update the `value` import on the Dask side, but I would still suggest we put a compat import back in `bokeh.models` for a little while longer.

**Comment by bryevdv** [comment]
@mattpap Dask has started CI testing with Bokeh dev versions, see https://github.com/dask/distributed/pull/6715 for a current status. 

TLDR: 

* I think we should add a module getattr for `bokeh.plotting.Figure` with a deprecation warning
* I think we should undo the rename of `field_name` to `field` for transforms (there is not a clean deprecation path AFAICT)
